PROTOCOL.md

```diff
@@ -2334,17 +2411,71 @@ content-type: application/x-ndjson; charset=utf-8
 Accepted timestamp format by a delta sharing server: in the ISO8601 format, in the UTC timezone, such as `2022-01-01T00:00:00Z`.   
 
-## API Response Format
+## Delta Sharing Capabilities Header
+This section explains the details of delta sharing capabilities header, which was introduced to help 
+delta sharing catch up with features in [delta protocol](https://github.com/delta-io/delta/blob/master/PROTOCOL.md).
+
+The key of the header is **delta-sharing-capabilities**, the value is semicolon separated capabilities. 
+Each capability is in the format of "key=value1,value2", values are separated by commas.
+Example: "responseformat=delta;readerfeatures=deletionvectors,columnmapping". All keys and values should
+be case-insensitive when processed by the server.
+
+This header can be used in the request for [Query Table Metadata](#query-table-metadata), 
+[Query Table](#read-data-from-a-table), and [Query Table Changes](#read-change-data-feed-from-a-table).
+
+**Compatibility**
+
+<table>
+<tr>
+<th>Client/Server</th>
+<th>Server that doesn't recognize the header</th>
+<th>Server that recognizes the header</th>
+</tr>
+<tr>
+<th>Client that doesn't recognize the header</th>
+<td>Response is in parquet format</td>
+<td>Response should be in parquet format.</td>
+</tr>
+<tr>
+<th>Client that recognizes the header</th>
+<td>The header is ignored at the server, and the format of the response should always be parquet.
+</td>
+<td>The header is processed properly by the server and the response is in the requested format.</td>
+</tr>
+</table>
+
+- If the client requested `delta` format and the response is in `parquet` format, the delta sharing
+client will NOT throw an error, the caller should decide the behavior (either to handle the response
+or to throw an error to let the user specify the format explicitly).
+- If the client requested `parquet` format and the response is in `delta` format, the delta sharing
+client should throw an error.
+
+### responseFormat
+Indicates the format to expect in the [API Response Format in Parquet](#api-response-format-in-parquet), two values are supported.
+
+- parquet: Represents the format of the delta sharing protocol that has been used in `delta-sharing-spark` 1.0 
+and less, also the default format if `responseFormat` is missing from the header, all the existing delta
+sharing connectors are able to process data in this format. 
+- delta: format can be used to read a shared delta table with minReaderVersion > 1, which contains 
+readerFeatures such as Deletion Vector or Column Mapping. `delta-sharing-spark` libraries 
+that are able to process `responseformat=delta` will be released soon.
+
+### readerFeatures
+readerfeatures is only useful when `responseformat=delta`, it includes values from [delta reader
+features](https://github.com/delta-io/delta/blob/master/PROTOCOL.md#table-features). It's set by the
+caller of `DeltaSharingClient` to indicate its ability to process delta readerFeatures.
```

Comment:
Question: should we make a table for all the supported capabilities header fields?

Reply:
I think it's up to the caller what capabilities are supported. The link to the full list is already added.